controllers/kafkamanagement/kafkauser_controller.go

```diff
@@ -533,3 +565,399 @@ func (r *KafkaUserReconciler) getKafkaUserCredsFromSecret(
 
 	return string(username[:len(username)-1]), string(password[:len(password)-1]), nil
 }
+
+func (r *KafkaUserReconciler) getKafkaUserCertIDFromSecret(
+	ctx context.Context,
+	certRequest *v1beta1.CertificateRequest,
+) (string, error) {
+	kafkaUserCertSecret := &v1.Secret{}
+	kafkaUserCertSecretNamespacedName := types.NamespacedName{
+		Name:      certRequest.SecretName,
+		Namespace: certRequest.SecretNamespace,
+	}
+
+	err := r.Get(ctx, kafkaUserCertSecretNamespacedName, kafkaUserCertSecret)
+	if err != nil {
+		return "", err
+	}
+
+	certID := kafkaUserCertSecret.Data["id"]
+
+	if len(certID) == 0 {
+		return "", models.ErrMissingSecretKeys
+	}
+
+	return string(certID), nil
+}
+
+func (r *KafkaUserReconciler) GenerateCSR(certRequest *v1beta1.CertificateRequest) (string, error) {
+	keyBytes, err := rsa.GenerateKey(rand.Reader, 1024)
```

Comment:
Let's use 2048 bit key for CSR generation - https://www.digicert.com/kb/csr-creation.htm#:~:text=General%20CSR%20Creation%20Guidelines&text=When%20you%20generate%20a%20CSR,(2048%2Dbit%20minimum).